client/framework/esx/compatibility.lua

```diff
@@ -30,10 +30,13 @@ RegisterNetEvent("skinchanger:getSkin", function(cb)
 end)
 
 RegisterNetEvent("skinchanger:loadSkin", function(skin, cb)
-    if not skin.hair_1 then
+    -- add validation invisible when failed registration (maybe server restarted when apply skin)
+    if skin.model then
         client.setPlayerAppearance(skin)
-        Framework.CachePed()
+    else
+        client.setPlayerAppearance(Config.DefaultSkin)
```

Comment:
Use `Config.InitialPlayerClothes`

Reply:
And check for the gender using `Framework.GetGender()` and apply the right skin based on that

Reply:
if i use `Framework.GetGender()` its make true already have skin

Reply:
You can use `Framework.GetGender(true)` to get either `Male` or `Female` based on the player data and not his skin. Then you can set it something like this:
```lua
client.setPlayerAppearance(Config.InitialPlayerClothes[Framework.GetGender(true)])
```

Reply:
wait test again

Reply:
> You can use `Framework.GetGender(true)` to get either `Male` or `Female` based on the player data and not his skin. Then you can set it something like this:
> 
> ```lua
> client.setPlayerAppearance(Config.InitialPlayerClothes[Framework.GetGender(true)])
> ```

still invis

Reply:
It doesn't work by default. Check the other comment where I mentioned that you have to modify Config.InitialPlayerClothes so that it stays backwards compatible and also works with setPlayerAppearance. If you can't figure it out then I can fix it myself and you can close the PR

Reply:
wait a sec

Reply:
> 

check my new commits please

Reply:
omg, i got typo

Reply:
ah already fixed